src/http/request_handler.rs

```diff
@@ -105,11 +105,22 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
     new_headers
 }
 
+pub async fn preflight(app_ctx: Arc<AppContext>, req: Request<Body>) -> Result<Response<Body>> {
+    let _whole_body = hyper::body::aggregate(req).await?;
+    let mut response = Response::builder()
+        .status(StatusCode::OK)
+        .body(Body::default())?;
+
+    update_response_headers(&mut response, app_ctx.as_ref());
```

Comment:
only cors related headers should be returned. add a test which has non cors + cors headers and response should have only cors related headers